docs/Technical Details/index.rst

```diff
@@ -0,0 +1,189 @@
+      tfenv:
+        aliases:
+        - tfe
+        allow_filter:
+        - match: exec
+          position: 0
+        file_extensions:
+        - .tf
+        security:
+          checkov:
+            command: checkov -d . --download-external-modules True --skip-download --output
+              json --output-file-path ${CHECKOV_JSON_REPORT_PATH}/checkov.json,
+            customizations:
+              CHECKOV_BASELINE: --baseline
+              CHECKOV_EXTERNAL_CHECKS_DIR: --external-checks-dir
+              CHECKOV_SKIP_CHECK: --skip-check
+            description: directory scan
+          kics:
+            command: kics scan --type Terraform --no-progress --queries-path ${KICS_INCLUDE_QUERIES_PATH}
+              --libraries-path ${KICS_LIBRARY_PATH} --report-formats json --output-path
+              ${KICS_JSON_REPORT_PATH} --output-name kics --path .
+            customizations:
+              KICS_EXCLUDE_SEVERITIES: --exclude-severities
+              KICS_INCLUDE_QUERIES: --include-queries
+            description: directory scan
+        version: v3.0.0
+        version_argument: --version
+
+After building ``easy_infra`` with this configuration, you should be able to expect that when you run ``tfenv exec init`` inside of an ``easy_infra`` container,
+then it would run both the ``kics`` and ``checkov`` commands as described under ``kics: command: ...`` and ``checkov: command: ...``, with additional
+customizations as defined under ``kics: customizations: ...`` and ``checkov: customizations: ...`` when the associated environment variables are set.
+
+As an example, if you ran ``tfenv exec init`` and also had the ``CHECKOV_BASELINE`` environment variable set to ``/iac/.checkov.baseline`` then the
+actual checkov command that would be run would be::
+
+    checkov -d . --download-external-modules True --skip-download --output json --output-file-path ${CHECKOV_JSON_REPORT_PATH}/checkov.json,
+    --baseline /iac/.checkov.baseline
+
+.. note::
+    The ``--baseline ...`` at the end was dynamically added due to the enviornment variable.
+
+Validation
+^^^^^^^^^^
+
+Sometimes security scanning tools are only equipped to run against IaC which is in a certain state, such as ensuing that the IaC is formatted properly and
+valid. ``validation`` is where you can specify what those are, and you can specify a list of commands to run in the specified order, prior to running the
+security scanning tools.
+
+Version
+^^^^^^^
+
+``version`` is where you can specify which versions of tool you want to include when you're buliding an ``easy_infra`` image. This is what is maintained by this
+project's automated maintenance scripts, and it is parsed into build arguments which are passed into the container image building process.
+
+Version Argument
+^^^^^^^^^^^^^^^^
+
+``version_argument`` is a way for us to describe how a command requests its version inside of ``easy_infra``. This is useful to know because we avoid running
```

Comment:
```suggestion
``version_argument`` is a way for us to describe how a command requests it's version inside of ``easy_infra``. This is useful to know because we avoid running
```